deepspeed/runtime/engine.py

```diff
@@ -981,6 +969,9 @@ def _configure_distributed_model(self, model):
                     hasattr(param,
                             'ds_id') for param in self.module.parameters()):
                 self.__check_params(self.module, torch.bfloat16)
+            if self.zero_optimization_stage() == 0 and not self.pipeline_parallelism:
+                raise NotImplementedError(
+                    "BF16 support is not yet implemented when not running ZeRO")
```

Comment:
Let's update this error to be more clear on what's supported/no-supported (talked offline)